### PR TITLE
Added nullability info to Page and Post.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Adds nullability information to `Page` and `Post`.

**To test:**
1. Make sure the app builds.
2. Make sure the tests build and run properly.
3. Go to the list of posts for a blog and interact with posts: search, filter, create, edit, view, view stats, trash, restore, delete.
4. Go to the list of pages for a blog and interact with posts: search, filter, create, edit, view, view stats, trash, restore, delete.

Needs review: @aerych

### CODE REVIEW

File: WordPress/Classes/Models/Post.h [hunk 1]
```diff
@@ -10,26 +12,26 @@ extern NSString * const PostTypeDefaultIdentifier;
...
-@property (nonatomic, strong) NSString *tags;
-@property (nonatomic, strong) NSString *postType;
-@property (nonatomic, strong) NSString *postFormat;
-@property (nonatomic, strong) NSString *postFormatText;
-@property (nonatomic, strong) NSSet *categories;
+@property (nonatomic, strong, nullable) NSNumber *commentCount;
+@property (nonatomic, strong, nullable) NSNumber *likeCount;
+@property (nonatomic, strong, nullable) Coordinate *geolocation;
+@property (nonatomic, strong, nullable) NSString *tags;
+@property (nonatomic, strong, nullable) NSString *postType;
```
Comment (kurzee) [inline WordPress/Classes/Models/Post.h]: `postType` is actually not an optional value in the CD model, and has a default value. So we could probably remove the `nullable` specifier.

**Comment by kurzee** [comment]
@diegoreymendez ending up reviewing this one since @aerych is out and I've touched these two files a couple times recently.

**Comment by kurzee** [comment]
@diegoreymendez all tests pass, everything looks good except the one nullability specifier, :shipit: unless you would like another set of 👀

**Comment by diegoreymendez** [comment]
@kurzee - Thanks for the review... I reassigned the task to you to avoid confusion.  I'll be reviewing your comments now.

File: WordPress/Classes/Models/Post.h (hunk 1 above)
Comment (diegoreymendez) [inline WordPress/Classes/Models/Post.h]: I've been thinking about this extensively, because I'm not sure the nullability info in CoreData can match the nullability info in our code in all cases.

Here's the reason: nullability in CoreData is only verified when saving changes (to the context), while nullability in Swift refers to the entire lifecycle of an object.

With Objective-C code, we won't see a problem setting it either way, because its runtime is very flexible and allows `nonnull` objects to be `nil`.

However if we migrated `Post` to Swift (which I'm tackling currently), I believe we would find out this property would need to be optional, since it doesn't have any initializer setting an initial value.

So long story short: would you mind me leaving this as optional until I can finish the migration to Swift?  It shouldn't be long and I'll make sure to /cc you on that PR.  If I can make this property non-optional I'll make sure to do it then.

Comment (kurzee) [inline WordPress/Classes/Models/Post.h]: Yep, makes sense 👍

**Comment by kurzee** [comment]
@diegoreymendez all good, :shipit:

**Comment by diegoreymendez** [comment]
Thanks @kurzee !
